jgiven-core/src/test/java/com/tngtech/jgiven/report/asciidoc/AsciiDocStepBlockConverterTest.java

```diff
@@ -0,0 +1,218 @@
+package com.tngtech.jgiven.report.asciidoc;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import com.tngtech.java.junit.dataprovider.DataProvider;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import com.tngtech.jgiven.annotation.Table;
+import com.tngtech.jgiven.report.ReportBlockConverter;
+import com.tngtech.jgiven.report.model.DataTable;
+import com.tngtech.jgiven.report.model.StepStatus;
+import com.tngtech.jgiven.report.model.Word;
+import java.util.Collections;
+import java.util.List;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(DataProviderRunner.class)
+public class AsciiDocStepBlockConverterTest {
+
+    private static final int ARBITRARY_DURATION = 3899;
+    private final ReportBlockConverter converter = new AsciiDocReportBlockConverter();
+
+    @Test
+    public void convert_step_without_description() {
+        // given
+        List<Word> words = ImmutableList.of(Word.introWord("given"), new Word("a coffee machine"));
+
+        // when
+        String block = converter.convertStepBlock(0, words, StepStatus.PASSED, ARBITRARY_DURATION, null, false);
+
+        // then
+        assertThatBlockContainsLines(block,
+            "* [.jg-intro-word]*Given* a coffee machine");
+    }
+
+    @Test
+    public void convert_step_with_description() {
+        // given
+        List<Word> words = ImmutableList.of(Word.introWord("given"), new Word("a coffee machine"));
+
+        // when
+        String block = converter.convertStepBlock(0, words, StepStatus.PASSED, ARBITRARY_DURATION,
+                "It is a brand new machine.", false);
+
+        // then
+        assertThatBlockContainsLines(block,
+            "* [.jg-intro-word]*Given* a coffee machine +",
+            "  _+++It is a brand new machine.+++_");
+    }
+
+    @Test
+    @DataProvider({"PASSED, check-square[role=green]", "FAILED, exclamation-circle[role=red]",
+        "SKIPPED, step-forward[role=silver]", "PENDING, ban[role=silver]"})
+    public void convert_step_from_unsuccessful_scenario_case(final StepStatus stepStatus, final String icon) {
```

Comment:
Given that the data provider also includes "passed", "unsuccesful" is inapproproate here. How about "converts_scenario_cases_step_status"?

Reply:
No, the unsuccessful does not described the status of the steps execution but the execution status of the scenario case it is included in.
In successful scenarios and scenario cases there is no need to print every step's execution status as they should all be `passed` only for non successful scenario cases the steps status should be reported.

Reply:
But that comment showed me that the status reporting was not implemented correctly for all parameter types ;(